substrate/frame/benchmarking/src/utils.rs

```diff
@@ -342,6 +342,44 @@ pub trait Benchmarking {
 	) -> Result<Vec<BenchmarkResult>, BenchmarkError>;
 }
 
+#[derive(Default)]
+pub struct BenchmarkRecording {
+	start_extrinsic: u128,
+	finish_extrinsic: u128,
```

Comment:
Maybe these should also be options so its impossible to call `elapsed_extrinsic` on default values?

Reply:
So you would panic in `fn elapsed_extrinsic` and `diff_pov` if the value have not been set?

Reply:
Yes, that should prevent misuse. Or return `Option` and then `expect` on the caller side.